code/game/atoms.dm

```diff
@@ -369,20 +369,21 @@
   * Otherwise it simply forceMoves the atom into this atom
   */
 /atom/proc/CheckParts(list/parts_list)
-	for(var/A in parts_list)
-		if(istype(A, /datum/reagent))
-			if(!reagents)
-				reagents = new()
-			reagents.reagent_list.Add(A)
-			reagents.conditional_update()
-		else if(ismovable(A))
-			var/atom/movable/M = A
-			if(isliving(M.loc))
-				var/mob/living/L = M.loc
-				L.transferItemToLoc(M, src)
-			else
-				M.forceMove(src)
-	parts_list.Cut()
+	if(parts_list)

```

Comment:
early return

also should probably lazylen this

Reply:
I considered doing `LAZYCLEARLIST(parts_list)` but then it would be checking the list twice for no reasons, on the `for` and then on the cutting.

Also considered doing an early return, but `CheckParts()` is a more general proc and I'd expect other things to be added to it. It felt more natural to add the check.

Sure you want the change? It's no trouble for me to make it.